spec/Classes.tex

```diff
@@ -784,12 +787,54 @@ \subsection{The Method Receiver and the {\em this} Argument}
 is \chpl{c1}.
 \end{chapelexample}
 
-The optional \sntx{this-intent} declares the intent by which the receiver
-argument will be passed.  When the value of \sntx{this-intent} is \chpl{param},
-it implies that the function can only be applied to param objects of the given
-type binding.  If the value is \chpl{ref}, the argument will be passed by
-reference---allowing modification of \chpl{this}.  Otherwise, the argument will
-be passed with the blank intent as specified in \rsec{The_Blank_Intent}.
+The optional \sntx{this-intent} declares how the receiver argument is
+constrained or will be passed for a given method call.
+
```

Comment:
If C++ permits static methods to be called on instances (which a quick test I wrote seems to indicate) then I think you're right that we shouldn't paint the analogy here (or else should change the implementation to match C++'s).  I'd apparently completely forgotten (or never realized) that this is the case.  Without looking, do you know whether purer OOP languages are similar in this respect?  I think I learned the concept of methods on classes rather than instances from Smalltalk (?) and assumed it was the same in C++, but am now wondering if I misunderstood Smalltalk as well (I was a lazy student).